src/attribution/onINP.ts

```diff
@@ -109,54 +104,55 @@ const saveInteractionTarget = (entry: PerformanceEventTiming) => {
 /**
  * Groups entries that were presented within the same animation frame by
  * a common `renderTime`. This function works by referencing
- * `pendingEntriesGroupMap` and using an existing render time if one is found
+ * `pendingEntriesGroups` and using an existing render time if one is found
  * (otherwise creating a new one). This function also adds all interaction
- * entries to an `entryToRenderTimeMap` WeakMap so that the "grouped" render
- * times can be looked up later.
+ * entries to an `entryToRenderTimeMap` WeakMap so that the "grouped" entries
+ * can be looked up later.
  */
 const groupEntriesByRenderTime = (entry: PerformanceEventTiming) => {
-  let renderTime = entry.startTime + entry.duration;
-  let previousRenderTime;
+  const renderTime = entry.startTime + entry.duration;
+  let group;
 
   latestProcessingEnd = Math.max(latestProcessingEnd, entry.processingEnd);
 
-  // Iterate of all previous render times in reverse order to find a match.
+  // Iterate over all previous render times in reverse order to find a match.
   // Go in reverse since the most likely match will be at the end.
-  for (let i = previousRenderTimes.length - 1; i >= 0; i--) {
-    previousRenderTime = previousRenderTimes[i];
-
-    // If a previous render time is within 8ms of the current render time,
-    // assume they were part of the same frame and re-use the previous time.
-    // Also break out of the loop because all subsequent times will be newer.
-    if (Math.abs(renderTime - previousRenderTime) <= 8) {
-      const group = pendingEntriesGroupMap.get(previousRenderTime)!;
+  for (let i = pendingEntriesGroups.length - 1; i >= 0; i--) {
+    const potentialGroup = pendingEntriesGroups[i];
+
+    // If a group's render time is within 8ms of the entry's render time,
+    // assume they were part of the same frame and add it to the group.
+    if (Math.abs(renderTime - potentialGroup.renderTime) <= 8) {
+      group = potentialGroup;
       group.startTime = Math.min(entry.startTime, group.startTime);
       group.processingStart = Math.min(
         entry.processingStart,
         group.processingStart,
       );
       group.processingEnd = Math.max(entry.processingEnd, group.processingEnd);
+      group.renderTime = Math.max(group.renderTime, renderTime);
```

Comment:
`previousRenderTimes` was never updated before like this, it was just the renderTime of whichever event was dispatched first, but it felt weird to not do this 🤷

Reply:
Or is it important that it's the first event's render time so groups are all entries with renderTime within 8ms of the first entry, not entries where they're all within 8ms of some other entry in that group? I can't really imagine a scenario where that could be a problem, but maybe if it's on a 144Hz monitor...

Reply:
Talking to @mmocny this could indeed (theoretically!) be problematic, so dropped it and maintaining the current behavior of using the renderTime of the first dispatched event in a group

Reply:
Yeah, I agree dropping it is the right call.